latch_cli/config/latch.py

```diff
@@ -5,53 +5,91 @@
-    "get-workflows": "/sdk/get-wf",
-    "verify": "/sdk/verify",
-    "remove": "/sdk/rm",
-    "id": "/sdk/node-id",
-    "mkdir": "/sdk/mkdir",
-    "rmdir": "/sdk/rmdir",
-    "touch": "/sdk/touch",
-    "pod-exec-info": "/sdk/get-pod-exec-info",
-    "provision-centromere": "/sdk/provision-centromere",
-    "get-image-from-task": "/sdk/get-image-from-task",
-    "preview": "/sdk/workflow-ui-preview",
-    "get-ws": "/sdk/get-ws",
-    "get-executions": "/sdk/get-executions",
-    "get-workflow-graph": "/sdk/get-workflow-graph",
-    "get-logs": "/sdk/get-logs-for-node",
-    "abort-execution": "/sdk/abort-execution",
-    "local-development": "/sdk/initiate-local-development-session",
-    "close-local-development": "/sdk/close-local-development-session",
-    "get-latest-version": "/sdk/get-latest-version-new",
-}
 
+@dataclass
+class _DataAPI:
+    begin_upload: str = "/sdk/initiate-multipart-upload"
+    complete_upload: str = "/sdk/complete-multipart-upload"
+    download: str = "/sdk/download"
+    id: str = "/sdk/node-id"
+    list: str = "/sdk/list"
+    remove: str = "/sdk/rm"
+    touch: str = "/sdk/touch"
+    mkdir: str = "/sdk/mkdir"
+    verify: str = "/sdk/verify"
+    test_data: str = "/sdk/get-test-data-creds"
 
+
+@dataclass
+class _WorkflowAPI:
+    upload_image: str = "/sdk/initiate-image-upload"
+    get_image: str = "/sdk/get-image-from-task"
+    register: str = "/sdk/register-workflow"
+    list: str = "/sdk/get-wf"
+    interface: str = "/sdk/wf-interface"
+    graph: str = "/sdk/get-workflow-graph"
+    check_version: str = "/sdk/check-workflow-version"
+    get_latest: str = "/sdk/get-latest-version-new"
+    preview: str = "/sdk/workflow-ui-preview"
+
+
+@dataclass
+class _ExecutionAPI:
+    create: str = "/sdk/wf"
+    list: str = "/sdk/get-executions"
+    abort: str = "/sdk/abort-execution"
+    logs: str = "/sdk/get-logs-for-node"
+    exec: str = "/sdk/get-pod-exec-info"
+
+
+@dataclass
+class _UserAPI:
+    jwt: str = "/sdk/access-jwt"
+    list_workspaces: str = "/sdk/get-ws"
+
+
+@dataclass
+class _CentromereAPI:
+    provision: str = "/sdk/provision-centromere"
+    start_local_dev: str = "/sdk/initiate-local-development-session"
+    stop_local_dev: str = "/sdk/close-local-development-session"
+
+
+@dataclass
+class _API:
+    data: _DataAPI
+    workflow: _WorkflowAPI
+    execution: _ExecutionAPI
+    user: _UserAPI
+    centromere: _CentromereAPI
+
+
+@dataclass
 class _LatchConfig:
-    dkr_repo = "812206152185.dkr.ecr.us-west-2.amazonaws.com"
-
-    def __init__(self):
-        self.console_url = CONSOLE_URL
-        self.nucleus_url = NUCLEUS_URL
-        self.sdk_endpoints = {
-            key: f"{self.nucleus_url}{endpoint}"
-            for key, endpoint in SDK_ENDPOINTS.items()
-        }
+    api: _API
+    dkr_repo: str = "812206152185.dkr.ecr.us-west-2.amazonaws.com"
+    console_url: str = CONSOLE_URL
+    nucleus_url: str = NUCLEUS_URL
+
+
+def build_endpoints(x: Type[T] = _API) -> T:
```

Comment:
shouldn't this be parameterized by `NUCLEUS_URL` not the type of the dataclass because thats the value thats changing, its cool but hard to read

Reply:
i mean it could, but i dont really see the point in reworking it, as this is

(1) not being used anywhere else,
(2) only being run once, and
(3) you realistically never need to touch this function, you can just add endpoints in the dataclasses directly and everything will be taken care of.

i dont think that its necessary also bc then id be needlessly duping work thats already in nucleus in a slightly different way